crates/provider/src/heart.rs

```diff
@@ -394,6 +405,9 @@ pub(crate) struct Heartbeat<S> {
     /// The stream of incoming blocks to watch.
     stream: futures::stream::Fuse<S>,
 
+    /// Lookbehind blocks in form of mapping block number -> vector of transaction hashes.
+    past_blocks: VecDeque<(u64, Vec<B256>)>,
```

Comment:
this should probably be a hashmap of hashes?

otherwise with >200txs per block we'll spend some time searching those vecs

Reply:
You mean `HashMap<u64, HashSet<B256>>`? 
I agree about `HashSet`, but for the wrapper IMHO we have a nice `pop_back` / `push_front` flow here, so probably I'd suggest making it `VecDeque<(u64, HashSet<B256>)>`. WDYT?

Reply:
Changed to `VecDeque<HashSet<(u64, HashSet<B256>)>`.